src/api/firebase.js

```diff
@@ -91,3 +91,21 @@ export async function createNewList(listId) {
 	}
 	return response;
 }
+
+export async function checkIfListExists(listId) {
+	const listCollectionRef = collection(db, listId);
+
+	try {
+		const querySnapshot = await getDocs(listCollectionRef);
+
+		// If the collection exists and has documents, return the snapshot
+		if (!querySnapshot.empty) {
+			return querySnapshot;
+		} else {
+			// Collection doesn't exist
+			return null;
+		}
+	} catch (error) {
+		return null;
+	}
```

Comment:
Sidenote: This would be where you send any error on the frontend to a service like Sentry or Datadog!